Add documentation for `ttnv3` source of `lorawan-stack-migrate` tool (#991)
TheThingsIndustries/lorawan-stack-docs: doc/content/getting-started/migrating/migrating-between-tts/use-migration-tool/_index.md

---
title: "Using Migration Tool"
description: ""
weight: 1
---

> This section explains how to migrate end devices between {{% tts %}} distributions.

Reviewer: ```suggestion
This section explains how to migrate end devices between {{% tts %}} distributions using the migration tool.
```

<!--more-->

> First, configure the migration tool with environment variables, or command-line arguments. See `--help` for more details:

Reviewer: ```suggestion
First, configure the migration tool with environment variables or command-line arguments, like shown below. See `--help` for more details.
```

```bash
$ export TTNV3_APP_ID="my-tts-app"                                                  # TTS App ID
$ export TTNV3_APP_API_KEY="NNSXS.U..."                                             # TTS App API Key (needs `device` permissions)
$ export TTNV3_APPLICATION_SERVER_GRPC_ADDRESS="eu1.cloud.thethings.network:8884"   # TTS Application Server URL Address
$ export TTNV3_IDENTITY_SERVER_GRPC_ADDRESS="eu1.cloud.thethings.network:8884"      # TTS Identity Server URL Address
$ export TTNV3_JOIN_SERVER_GRPC_ADDRESS="eu1.cloud.thethings.network:8884"          # TTS Join Server URL Address
$ export TTNV3_NETWORK_SERVER_GRPC_ADDRESS="eu1.cloud.thethings.network:8884"       # TTS Network Server URL Address
$ export TTNV3_CA_FILE="/path/to/ca.file"                                           # Path to a CA file (optional)
```

{{< note >}} The export process will halt if any error occurs. Execute commands with the `--dry-run` flag to verify whether the outcome will be as expected. {{</ note >}}

> Then proceed to exporting devices or applications:

Reviewer: ```suggestion
Then proceed to exporting devices or applications.
```

{{< tabs/container "Export devices" "Export applications" >}}

{{< tabs/tab "Export devices" >}}

> To export a single device using it's **DeviceID** (e.g. `my-device`):

Reviewer: ```suggestion
To export a single device using its **End device ID** (e.g. `my-device`):
```

Reviewer: This is just to line it up with names in the Console

```bash
# dry run first, verify that no errors occur
$ ttn-lw-migrate device --source ttnv3 'my-device' --dry-run --verbose > devices.json
# export device
$ ttn-lw-migrate device --source ttnv3 'my-device' > devices.json
```

In order to export a large number of devices, create a file named `device_ids.txt` with one device ID per line:

```
my-device-1
my-device-2
my-device-3
```

And then export with:

```bash
# dry run first, verify that no errors occur
$ ttn-lw-migrate devices --source ttnv3 --dry-run --verbose < device_ids.txt > devices.json
# export devices
$ ttn-lw-migrate devices --source ttnv3 < device_ids.txt > devices.json
```

{{</ tabs/tab >}}

{{< tabs/tab "Export applications" >}}

> To export all devices of an application using it's **ApplicationID** (e.g. `my-app-id`):

Reviewer: ```suggestion
To export all devices of an application using its **Application ID** (e.g. `my-app-id`):
```
